### PR TITLE
refactor: MapData file structure refactoring

### PR DESCRIPTION
**Pull request by magicus**
This started out by my getting annoyed by the fact that "Json" did not mean json but just a generic trivial class implementation of our interfaces. After renaming them to "Impl" I had to move them to somewhere better, and after that I realized the entire structure needed some extra love and care.

I don't think I'm 100% happy with this layout, but it least it is far superior to the current one. I suggest we use this for the time being, and then we can complement it with additional refactorings when the last pieces starts to fall in place.

### CODE REVIEW

File: common/src/main/java/com/wynntils/services/mapdata/attributes/impl/AbstractMapAttributes.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public Optional<MapDecoration> getIconDecoration() {
     }
 
     @Override
-    public Optional<MarkerOptions> getMarkerOptions() {
+    public Optional<MapMarkerOptions> getMarkerOptions() {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/services/mapdata/attributes/impl/AbstractMapAttributes.java:67]: MapMarkerOptions? Technically it's also in world markers, in fact, it mostly is.

**Review by kristofbolyai — APPROVED** [review]
I'll merge this as is, huge refactors like this is a merging nightmare if we do anything with mapdata. However, I think we have to find a new name for `MapMarkerOptions`.

File: common/src/main/java/com/wynntils/services/mapdata/attributes/impl/AbstractMapAttributes.java (hunk 1 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/services/mapdata/attributes/impl/AbstractMapAttributes.java:67]: I see your point. And sure, we can change it back if you really want. My thinking was to try to get a common prefix for all "map data". And even though we display it in the "world", it is still map data. I mean, you could technically tint the world the same color as the color of the territory you are currently in. So to me "map data" means like "geography-based information"; but it does not strictly have to be displayed on a map. It is also used in the `/locate` command, etc.

So maybe we should call it `GeoData` instead? 🤔

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/services/mapdata/attributes/impl/AbstractMapAttributes.java:67]: I'm not sure if that's better 😅

Comment (magicus) [reply common/src/main/java/com/wynntils/services/mapdata/attributes/impl/AbstractMapAttributes.java:67]: Or we can use the precedence from GIS and call it ["geospatial data"](https://en.wikipedia.org/wiki/Geographic_data_and_information). ;-)

Or we stick to the "Map" but are clear (perhaps by editing the glossary wiki) that this term refers to geospatial data, which are not strictly necessarily displayed on a map (even if that is the primary use case).

**Pull request merged**